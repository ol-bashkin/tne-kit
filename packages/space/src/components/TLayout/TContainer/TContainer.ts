import { computed, h, defineComponent } from "vue";

export default defineComponent({
  name: "TContainer",

  props: {
    size: {
      type: [Boolean, String],
      default: "fluid",
    },
    tag: {
      type: String,
      default: "div",
    },
  },

  setup(props, { slots }) {
    const classes = computed(() => {
      const classList: any[] = [];

      classList.push({
        [`container-${props.size}`]: props.size,
      });

      return classList;
    });

    return () =>
      h(
        props.tag,
        {
          class: ["t-container", classes.value],
        },
        slots.default?.()
      );
  },
});
