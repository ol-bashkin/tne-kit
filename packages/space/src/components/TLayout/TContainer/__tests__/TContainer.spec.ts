import { mount } from "@vue/test-utils";
import TContainer from "../TContainer";

describe("TContainer.ts", () => {
  test("should have an t-container class", () => {
    const wrapper = mount(TContainer, {
      propsData: {
        size: "lg",
      },
    });
    expect(wrapper.classes()).toContain("t-container");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
