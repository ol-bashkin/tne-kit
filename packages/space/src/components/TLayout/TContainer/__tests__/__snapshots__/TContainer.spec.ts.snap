// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TContainer.ts should have an t-container class 1`] = `<div class="t-container container-lg"></div>`;
